### PR TITLE
Prepare for 44.0.0 release: version and changelog

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

- Part of https://github.com/apache/datafusion/issues/13334

## Rationale for this change
Prepare for the release

## What changes are included in this PR?
1. Update version
2. add changelog

### CODE REVIEW

**Review by phillipleblanc — APPROVED** [review]
🚀

File: dev/release/generate-changelog.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def generate_changelog(repo, repo_name, tag1, tag2, version):
     unique_pulls = []
     all_pulls = []
     for commit in comparison.commits:
+        print(".", file=sys.stderr, end=" ", flush=True) # progress indicator
```
Comment (goldmedal) [inline dev/release/generate-changelog.py:47]: 👍

**Review by goldmedal — APPROVED** [review]
🚀

File: dev/release/generate-changelog.py (hunk 1 above)
Comment (alamb) [reply dev/release/generate-changelog.py:47]: Yeah, sorry I should have commented on that one -- it is due to the fact I am on a relatively slow internet connection and it wasn't clear to me the script was actually doing anything without this feedback.

**Comment by alamb** [comment]
Thank you @phillipleblanc  and @goldmedal 

I am waiting on the final ticket required for the 44 release:
- https://github.com/apache/datafusion/issues/13334#issuecomment-2561104600

Then I will update the changelog with the most recent commits, merge this PR and make a release candidate

**Pull request merged**